### DISCUSSION TITLE
Firing offset from the origin of turrets.

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#4756] [Ideas and suggestions]
At the moment, the "barrel" property of turrets functions as the origin point that the turret rotates around and fires from rather than how it does for characters' ranged-weapons.

I am suggesting the ability to offset where a projectile fires from in relation to the turret's center rotation through either a direct 1-dimentional radius or vector - as is the case with ranged weapons.
![image](https://user-images.githubusercontent.com/55356135/104634868-2eb67480-566f-11eb-83ec-5d63e60067d0.png)

One purpose could be for allowing structures to be placed near / inside the origin of a turret without risk of collision.
![image](https://user-images.githubusercontent.com/55356135/104635989-af29a500-5670-11eb-82c7-4462c5447352.png)

**Comment by DeinococcusR** [comment]
I totally support this. Had my own problems regarding this recently. It's especially annoying if you have turrets with a longer barrels(like shown in EKs second image).
